clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1361,16 +1378,21 @@ public boolean isDone() {
         private final Callback userCallback;
         private final ProducerInterceptors<K, V> interceptors;
         private final TopicPartition tp;
+        private final Time time;
+        private final long startTime;
 
-        private InterceptorCallback(Callback userCallback, ProducerInterceptors<K, V> interceptors, TopicPartition tp) {
+        private InterceptorCallback(Callback userCallback, long startTime, Time time, ProducerInterceptors<K, V> interceptors, TopicPartition tp) {
```

Comment:
though not used in other places, it's better to add another constructor with additional params and let the original constructor invoke the new one, to avoid breaking backward compatibility.

Reply:
Addressed.